lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -52,17 +54,47 @@ struct AffineToPipeline : public AffineToPipelineBase<AffineToPipeline> {
   void runOnOperation() override;
 
 private:
+  ModuloProblem getModuloProblem(CyclicProblem &prob);
   LogicalResult
   lowerAffineStructures(MemoryDependenceAnalysis &dependenceAnalysis);
-  LogicalResult populateOperatorTypes(SmallVectorImpl<AffineForOp> &loopNest);
-  LogicalResult solveSchedulingProblem(SmallVectorImpl<AffineForOp> &loopNest);
-  LogicalResult createPipelinePipeline(SmallVectorImpl<AffineForOp> &loopNest);
+  LogicalResult populateOperatorTypes(SmallVectorImpl<AffineForOp> &loopNest,
+                                      ModuloProblem &problem);
+  LogicalResult solveSchedulingProblem(SmallVectorImpl<AffineForOp> &loopNest,
+                                       ModuloProblem &problem);
+  LogicalResult createPipelinePipeline(SmallVectorImpl<AffineForOp> &loopNest,
+                                       ModuloProblem &problem);
 
   CyclicSchedulingAnalysis *schedulingAnalysis;
 };
 
 } // namespace
 
+ModuloProblem AffineToPipeline::getModuloProblem(CyclicProblem &prob) {
+  auto modProb = ModuloProblem::get(prob.getContainingOp());
+  for (auto *op : prob.getOperations()) {
+    auto opr = prob.getLinkedOperatorType(op);
+    if (opr.has_value()) {
```

Comment:
At the point `getModuloProblem` is called, I don't think we actually have any operator types yet do we?